src/lightning/pytorch/trainer/supporters.py

```diff
@@ -217,171 +95,91 @@ class CombinedLoader:
         {'a': tensor([4, 5]), 'b': tensor([5, 6, 7, 8, 9])}
     """
 
-    def __init__(self, loaders: Any, mode: str = "min_size"):
+    def __init__(self, loaders: Any, mode: _LITERAL_SUPPORTED_MODES = "min_size") -> None:
         if mode not in _supported_modes:
             raise ValueError(f"Unsupported mode {mode!r}, please select one of: {list(_supported_modes)}.")
+        self._loaders = loaders
+        self._loaders_flattened, self._loaders_spec = _tree_flatten(loaders)
 
-        self.loaders = loaders
-
-        datasets = apply_to_collection(
-            self.loaders, Iterable, getattr, "dataset", None, wrong_dtype=(Sequence, Mapping)
+        # FIXME(carlos): why do we even need this?
```

Comment:
This is something I want to explore, but not for this PR.

Reply:
Is this then fixme (needs to be done before merge) or rather todo (can be done later?)

Reply:
Changed to TODO